### PR TITLE
Expose swarmkit's Raft tuning parameters in engine config

### PR DESCRIPTION
**Pull request by chungers**
Signed-off-by: David Chung <david.chung@docker.com>

**- What I did**
This PR exposes the `HeartbeatTick` and `ElectionTick` parameters for Swarmkit's Raft quorum.  This allows tuning of the leader election behavior of the Swarm managers.  This PR contains default values so these settings are strictly optional.  Changing these settings can adjust the sensitivity of Swarm managers to transient events in their environment and make the quorum more stable.

**- How I did it**
Swarmkit already exposes these parameters but they are previously hardcoded.  This PR plumbs these settings through to expose them in the daemon's config.

**- How to verify it**
Changing the daemon's config JSON should override the default values.

**- Description for the changelog**

Make Swarm manager Raft quorum parameters configurable in daemon config.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by boaz0 — APPROVED** [review]
LGTM :+1:

**Review by vdemeester — APPROVED** [review]
SGTM

**Review by anshulpundir — APPROVED** [review]

**Comment by stevvooe** [comment]
LGTM

**Comment by tonistiigi** [comment]
LGTM

**Comment by thaJeztah** [comment]
Looks like this needs an update in the docs, also;

- should this be exposed as a flag as well?
- is this configuration live-reloadable?

File: daemon/cluster/cluster.go [hunk 1]
```diff
@@ -134,6 +141,14 @@ func New(config Config) (*Cluster, error) {
 	if config.RuntimeRoot == "" {
 		config.RuntimeRoot = root
 	}
+	if config.RaftHeartbeatTick == 0 {
+		config.RaftHeartbeatTick = 1
+	}
+	if config.RaftElectionTick == 0 {
+		// 10X heartbeat tick is the recommended ratio according to etcd docs.
+		config.RaftElectionTick = 10 * config.RaftHeartbeatTick
```
Comment (thaJeztah) [inline daemon/cluster/cluster.go]: What happens if I configure `RaftElectionTick < RaftHeartbeatTick`? Should `RaftElectionTick` always be multiple times `RaftHeartbeatTick`? If so; should we add validation here, or have a `ratio` instead as configuration?

Comment (chungers) [reply daemon/cluster/cluster.go]: This here doesn't enforce that `RaftElectionTick` should always be > than `RaftHearbeatTick` and should be at least 10 (which is the default value when nothing is set).  

I can add a check to return an error when `RaftElectionTick` is strictly greater than `RaftHeartbeatTick` and prevent the daemon from starting up.  Is this acceptable?

**Pull request merged**